services/core/java/com/android/server/locksettings/LockSettingsService.java

```diff
@@ -1732,6 +1732,142 @@ public boolean setLockCredential(LockscreenCredential credential,
         }
     }
 
+
+    private void enforceCallerPermisson() {
+        if (!hasPermission(PERMISSION) && !hasPermission(SET_AND_VERIFY_LOCKSCREEN_CREDENTIALS)) {
+            throw new SecurityException(
+                    "validDuressPinExist requires SET_AND_VERIFY_LOCKSCREEN_CREDENTIALS or "
+                            + PERMISSION);
+        }
+    }
+
+    @Override
+    public boolean validDuressCredentialsExist() {
+        enforceCallerPermisson();
+        return mStorage.readDuressPinToken() != null
+                && mStorage.readDuressPinToken().length > 0
+                && mStorage.readDuressPasswordToken() != null
+                && mStorage.readDuressPasswordToken().length > 0;
+    }
+
+    @Override
+    public void deleteDuressConfig() {
```

Comment:
There's code that handles deleting duress config in `setDuressCredentialInternal`, and it requires the user credential. Shouldn't it be used instead of this method?